src/main/java/com/airlines/dao/AirplaneDaoImpl.java

```diff
@@ -0,0 +1,189 @@
+            return fillStatementWithAccountData(insertStatement, airplane);
+        } catch (SQLException e) {
+            throw new DaoOperationException("Cannot prepare statement to insert Airplane", e);
+        }
+    }
+
+    private PreparedStatement fillStatementWithAccountData(PreparedStatement insertStatement, Airplane airplane)
+            throws SQLException {
+        insertStatement.setString(1, airplane.getCodeName());
+        insertStatement.setString(2, airplane.getCodeName());
+        insertStatement.setString(2, String.valueOf(airplane.getModel()));
+        insertStatement.setDate(4, Date.valueOf(airplane.getManufacture()));
+        insertStatement.setInt(5, airplane.getCapacity());
+        insertStatement.setInt(5, airplane.getFlightRange());
+        insertStatement.setInt(5, airplane.getCrews_id());
+        return insertStatement;
+    }
+
+    private void executeUpdate(PreparedStatement insertStatement, String errorMessage) throws SQLException {
+        int rowsAffected = insertStatement.executeUpdate();
+        if (rowsAffected == 0) {
+            throw new DaoOperationException(errorMessage);
+        }
+    }
+
+    private int fetchGeneratedId(PreparedStatement insertStatement) throws SQLException {
+        ResultSet generatedKeys = insertStatement.getGeneratedKeys();
+        if (generatedKeys.next()) {
+            return generatedKeys.getInt(1);
+        } else {
+            throw new DaoOperationException("Can not obtain an Airplane id");
+        }
+    }
+
+    @Override
+    public Airplane findOne(String codeName) {
+        try (Connection connection = dataSource.getConnection()) {
+            return findAirplaneByCodeName(codeName, connection);
+        } catch (SQLException e) {
+            throw new DaoOperationException(String.format("Cannot find Airplane by codeName = %s", codeName), e);
+        }
+    }
+
+    private Airplane findAirplaneByCodeName(String codeName, Connection connection) throws SQLException {
+        PreparedStatement selectByIdStatement = prepareSelectByCodeNameStatement(codeName, connection);
+        ResultSet resultSet = selectByIdStatement.executeQuery();
+        resultSet.next();
+        return parseRow(resultSet);
+    }
+
+    private PreparedStatement prepareSelectByCodeNameStatement(String codeName, Connection connection) {
+        try {
+            PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_AIRPLANE_BY_ID_SQL);
+            selectByIdStatement.setString(1, codeName);
+            return selectByIdStatement;
+        } catch (SQLException e) {
+            throw new DaoOperationException("Cannot prepare statement to select airplane by codeName", e);
+        }
+    }
+
+    private Airplane parseRow(ResultSet resultSet) throws SQLException {
+        return Airplane.builder()
+                .withId(resultSet.getInt("id"))
+                .withCodeName(resultSet.getString("codeName"))
```

Comment:
compare with https://github.com/vitaly-lab/airlines-app/blob/feature/VAA-4/src/main/java/com/airlines/dao/CrewDaoImpl.java

code_name instead codeName, resultSet contains database table row

Reply:
fixed in [3d2fa27](https://github.com/vitaly-lab/airlines-app/pull/6/commits/3d2fa27df397dda1795b8b92332e2576be3067c6)